src/realm/db.cpp

```diff
@@ -1147,7 +1147,9 @@ void DB::do_open(const std::string& path, bool no_create_file, bool is_backend,
                 info->number_of_versions = 1;
 
                 info->latest_version_number = version;
-                alloc.m_youngest_live_version = version;
+                // there isn't really any old mappings to purge, but this has the side effect of properly
+                // setting up the memory mapping machinery.
+                alloc.purge_old_mappings(version, version);
```

Comment:
Seems hacky, since this comment is repeated in multiple places. Would it make sense to extract to a separate method?